editor/test/editor/graphics/types_test.clj

```diff
@@ -0,0 +1,185 @@
+;;
+;; Unless required by applicable law or agreed to in writing, software distributed
+;; under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+;; CONDITIONS OF ANY KIND, either express or implied. See the License for the
+;; specific language governing permissions and limitations under the License.
+
+(ns editor.graphics.types-test
+  (:require [clojure.test :refer :all]
+            [editor.graphics.types :as graphics.types]
+            [editor.protobuf :as protobuf])
+  (:import [com.dynamo.graphics.proto Graphics$CoordinateSpace Graphics$VertexAttribute$DataType Graphics$VertexAttribute$SemanticType Graphics$VertexAttribute$VectorType Graphics$VertexStepFunction]))
+
+(set! *warn-on-reflection* true)
+
+(def ^:private random-values [nil 1 "string" :keyword (Object.) #{}])
+
+(defn- pb-ints-by-val [^Class pb-enum-class]
+  (into {}
+        (map (juxt protobuf/pb-enum->val protobuf/pb-enum->int))
+        (protobuf/protocol-message-enums pb-enum-class)))
+
+(deftest buffer-data-type-data-type-test
+  (is (= {:byte :type-byte
+          :ubyte :type-unsigned-byte
+          :short :type-short
+          :ushort :type-unsigned-short
+          :int :type-int
+          :uint :type-unsigned-int
+          :float :type-float}
+         (into {}
+               (map (juxt identity graphics.types/buffer-data-type-data-type))
+               (disj editor.buffers/buffer-data-types :double :long)))))
+
+(deftest coordinate-space?-test
+  (is (every? false? (map graphics.types/coordinate-space? random-values)))
+  (is (every? true? (map graphics.types/coordinate-space? graphics.types/coordinate-spaces)))
+  (is (true? (graphics.types/coordinate-space? :coordinate-space-default))))
+
+(deftest concrete-coordinate-space?-test
+  (is (every? false? (map graphics.types/concrete-coordinate-space? random-values)))
+  (is (every? true? (map graphics.types/concrete-coordinate-space? graphics.types/concrete-coordinate-spaces)))
+  (is (false? (graphics.types/concrete-coordinate-space? :coordinate-space-default))))
+
+(deftest coordinate-space-pb-int-test
+  (is (= (pb-ints-by-val Graphics$CoordinateSpace)
+         (into {}
+               (map (juxt identity graphics.types/coordinate-space-pb-int))
+               graphics.types/coordinate-spaces))))
+
+(deftest data-type?-test
+  (is (every? false? (map graphics.types/data-type? random-values)))
+  (is (every? true? (map graphics.types/data-type? graphics.types/data-types))))
+
+(deftest data-type-byte-size-test
+  (is (= {:type-byte Byte/BYTES
+          :type-unsigned-byte Byte/BYTES
+          :type-short Short/BYTES
+          :type-unsigned-short Short/BYTES
+          :type-int Integer/BYTES
+          :type-unsigned-int Integer/BYTES
+          :type-float Float/BYTES}
+         (into {}
+               (map (juxt identity graphics.types/data-type-byte-size))
+               graphics.types/data-types))))
+
+(pb-ints-by-val Graphics$CoordinateSpace)
```

Comment:
Leftover form?

Reply:
Hmm, yes. Honestly, I left a bit of a mess all around this PR. Not sure what happened!